src/Compilers/Core/Portable/Operations/OperationFactory.cs

```diff
@@ -1,29 +1,33 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Semantics
 {
     internal static class OperationFactory
     {
-        public static IVariableDeclaration CreateVariableDeclaration(ILocalSymbol variable, IOperation initialValue, SemanticModel semanticModel, SyntaxNode syntax)
+        public static IVariableDeclaration CreateVariableDeclaration(ILocalSymbol variable, IVariableInitializer initializer, SemanticModel semanticModel, SyntaxNode syntax)
         {
-            return CreateVariableDeclaration(ImmutableArray.Create(variable), initialValue, semanticModel, syntax);
+            return CreateVariableDeclaration(ImmutableArray.Create(variable), initializer, semanticModel, syntax);
         }
 
-        public static VariableDeclaration CreateVariableDeclaration(ImmutableArray<ILocalSymbol> variables, IOperation initialValue, SemanticModel semanticModel, SyntaxNode syntax)
+        public static VariableDeclaration CreateVariableDeclaration(ImmutableArray<ILocalSymbol> variables, IVariableInitializer initializer, SemanticModel semanticModel, SyntaxNode syntax)
         {
             return new VariableDeclaration(
                 variables,
-                initialValue,
+                initializer,
                 semanticModel,
                 syntax,
                 type: null,
                 constantValue: default(Optional<object>),
                 isImplicit: false); // variable declaration is always explicit
         }
 
+        public static IVariableInitializer CreateVariableInitializer(SyntaxNode syntax, IOperation initializerValue, SemanticModel semanticModel, bool isImplicit)
+        {
+            return new VariableInitializer(initializerValue, semanticModel, syntax, initializerValue.Type, constantValue: default, isImplicit: isImplicit);
```

Comment:
>initializerValue.Type [](start = 84, length = 21)

I think IVariableInitializer shouldn't have type, it is not an expression.

Reply:
Makes sense, let me change and update the test baselines.